manila/scheduler/drivers/filter.py

```diff
@@ -259,6 +260,15 @@ def _schedule_share(self, context, request_spec, filter_properties=None):
         # host for the job.
         weighed_hosts = self.host_manager.get_weighed_hosts(hosts,
                                                             filter_properties)
+
+        # Call an external service that can modify `weighed_hosts` once more.
+        # This service may filter out some hosts, or it may re-order them.
+        weighed_hosts = call_external_scheduler_api(
+            context, weighed_hosts, request_spec)
+        if not weighed_hosts:
+            msg = "No valid hosts found after calling external scheduler API."
+            raise exception.NoValidHost(reason=msg)
```

Comment:
I wonder if a generic user message would still be created, if we don't speficy the `detail_data` next to the `reason`

At least when we really start filtering and not just do weighing, we should consider providing the details here, so users (and not only admins) would get an idea why scheduling failed